### PR TITLE
feat: [VRD-817] Implement RegisteredModelVersion.list_builds()

### PR DESCRIPTION
**Pull request by liuverta**
## Impact and Context

Now a user can list all the builds of a model version!

## Risks and Area of Effect

None; entirely new functionality, covered by tests.

## Testing
- [x] Unit test
- [ ] Deployed to dev env
- [ ] Other (explain) 

In Python 3.10:

```
% pytest unit_tests/deployment/test_build.py deployment/test_build.py 
==================================================== test session starts ====================================================
platform darwin -- Python 3.10.6, pytest-7.2.1, pluggy-1.0.0
rootdir: /Users/miliu/Documents/modeldb/client/verta/tests, configfile: pytest.ini
plugins: forked-1.4.0, xdist-3.1.0, typeguard-2.13.3, hypothesis-6.67.1
collected 5 items                                                                                                           

unit_tests/deployment/test_build.py ...                                                                               [ 60%]
deployment/test_build.py ..                                                                                           [100%]

============================================== 5 passed, 3 warnings in 57.14s ===============================================
```

## Reverting
- [ ] Contains Migration - _Do Not Revert_

Revert this PR.

### CODE REVIEW

**Comment by liuverta** [comment]
!rebuild

File: client/verta/verta/registry/entities/_modelversion.py [hunk 1]
```diff
@@ -1690,3 +1692,38 @@ def del_dataset_version(self, key):
...
+        .. code-block:: python
+
+            from datetime import datetime, timedelta
+
+            past_builds = list(filter(
+                lambda build: build.get_scan().date_updated < datetime.now().astimezone() - timedelta(days=30),
+                model_ver.list_builds(),
+            ))
+
+        """
```
Comment (liuverta) [inline client/verta/verta/registry/entities/_modelversion.py:1728]: Screen Shot 2023-04-20 at 4 14 16 PM

File: client/verta/verta/endpoint/_endpoint.py [hunk 2]
```diff
@@ -11,8 +11,8 @@
 
 from verta.deployment import DeployedModel
 from verta._internal_utils import _utils, arg_handler, kafka
-from verta.tracking.entities import ExperimentRun
-from verta.registry.entities import RegisteredModelVersion
+import verta.tracking.entities as tracking_entities
+import verta.registry.entities as registry_entities
```
Comment (liuverta) [inline client/verta/verta/endpoint/_endpoint.py:15]: This (and the rest of this file) is just to resolve a circular import ☹️

File: client/verta/verta/registry/entities/_modelversion.py (hunk 1 above)
Comment (ghost) [reply client/verta/verta/registry/entities/_modelversion.py:1728]: it's cool we can do all this filtering now!!

File: client/verta/tests/deployment/test_build.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+from verta.environment import Python
+
+
+def test_endpoint_get_current_build(model_version, endpoint, model_for_deployment):
+    """Verify get_current_build() returns expected build."""
+    model_version.log_model(model_for_deployment["model"], custom_modules=[])
+    model_version.log_environment(Python([model_for_deployment["requirements"].read()]))
+
+    build = endpoint._create_build(model_version)
+    endpoint.update(build, wait=False)
```
Comment (ghost) [inline client/verta/tests/deployment/test_build.py:12]: didn't know we could do this

**Review by ghost — APPROVED** [review]

**Comment by liuverta** [comment]
I didn't either until I started working on this ahaha

File: client/verta/verta/registry/entities/_modelversion.py (hunk 1 above)
Comment (ewagner-verta) [reply client/verta/verta/registry/entities/_modelversion.py:1728]: Nice work on these examples!

**Review by ewagner-verta — APPROVED** [review]

**Pull request merged**